ingenialink/ethercat/network.py

```diff
@@ -516,23 +519,27 @@ def _notify_status(self, slave_id: int, status: NET_DEV_EVT) -> None:
         for callback in self.__observers_net_state[slave_id]:
             callback(status)
 
-    def _recover_from_power_cycle(self, status: NET_DEV_EVT) -> None:
-        """Transition the slaves to Pre-Op state after a power cycle.
+    def _recover_from_disconnection(self) -> bool:
+        """Recover the CoE communication after a disconnection.
+        All the connected slaves need to transitioned to the PreOp state.
 
-        Args:
-            status: The network status.
-
-        Raises:
-            ILStateError: If not all slaves reach PreOp state.
+        Returns:
+            True if all the connected slaves reach the PreOp state.
 
         """
         self._ecat_master.read_state()
-        if status == NET_DEV_EVT.ADDED and self._ecat_master.state == pysoem.INIT_STATE:
-            self.__init_nodes()
-            if not self._check_node_state(self.servos, pysoem.PREOP_STATE):
-                raise ILStateError(
-                    "The communication cannot be recovered. Not all slaves reached PreOp state"
-                )
+        if self._ecat_master.state > pysoem.INIT_STATE:
+            return True
+        self.__init_nodes()
+        all_drives_in_preop = self._check_node_state(self.servos, pysoem.PREOP_STATE)
```

Comment:
servos are only the ones you are connected to right?
This is compatible with servos in BOOT state on the network, right?

Reply:
A slave cannot be connected to the network if it cannot reach the PreOp state.

Reply:
by connected I mean "ingenialink connected".
We should be able to have a network of two drives:
- One in app, can reach PreOp -> Can connect to it.
- One in boot, cannot reach preop -> Can coexist in the network but we cannot connect to it

Reply:
I don't think we should have a servo in the network if we cannot interact with it.  In any case, this subject is out of the scope of this issue.

Reply:
As long as it doesn't alter the current behaviour regarding this is fine